### PR TITLE
Adds metadata lineage page

### PR DESCRIPTION
**Pull request by antony-britto**
## Description
Migrating metadata lineage page from angular to react

## PR Type
- [ ] Bug Fix
- [X] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement

## Links
Jira: [18434](https://cdap.atlassian.net/browse/CDAP-18434)

## Test Plan
Added unit tests

## Screenshots
image

### CODE REVIEW

File: app/cdap/components/Metadata/Lineage/LineageDiagram/index.tsx [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+const DAG_CONTAINER_ID = `dag-${uuidV4()}`;
+
+const NoContentMsg = styled.p`
+  margin-top: 150px;
+  text-align: center;
+  font-size: 1.39em;
+  color: #666;
+`;
+
+const LineageContainer = styled.div`
```
Comment (njbriggs) [inline app/cdap/components/Metadata/Lineage/LineageDiagram/index.tsx:38]: Can you split the child styles up into their own styled components?

Comment (antony-britto) [reply app/cdap/components/Metadata/Lineage/LineageDiagram/index.tsx:38]: @njbriggs ignore this PR for now.

File: app/cdap/components/Metadata/Lineage/ProgramModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
```
Comment (seanfeld) [inline app/cdap/components/Metadata/Lineage/ProgramModal/index.tsx]: 22

File: app/cdap/components/Metadata/Lineage/__tests__/helper.test.ts [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+      entityId: 'logs_data_sink',
+      entityType: 'datasets',
+      levels: 1,
+      namespace: 'default',
+      rollup: 'workflow',
+      start: 'now-365d',
+    };
+
+    it('return parsed response for given lineage response', async () => {
+      //   MySearchApi.getDatasetProperties.mockResolvedValue({});
```
Comment (seanfeld) [inline app/cdap/components/Metadata/Lineage/__tests__/helper.test.ts]: do we need this comment?

**Review by seanfeld — APPROVED** [review]
Looks good - I just have one nit: some of the files are pretty big

**Review by njbriggs — APPROVED** [review]

**Review by mrahanjam — APPROVED** [review]
LGTM!

File: app/cdap/components/Metadata/Lineage/ProgramModal/index.tsx (hunk 2 above)
Comment (antony-britto) [reply app/cdap/components/Metadata/Lineage/ProgramModal/index.tsx]: Done

File: app/cdap/components/Metadata/Lineage/__tests__/helper.test.ts (hunk 3 above)
Comment (antony-britto) [reply app/cdap/components/Metadata/Lineage/__tests__/helper.test.ts]: Removed

**Comment by antony-britto** [comment]
> Looks good - I just have one nit: some of the files are pretty big

Moved styles & constants into a separate file to address this.

**Pull request merged**